src/components/graph-controls.js

```diff
@@ -1,155 +1,93 @@
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
 
-const steps = 100; // Slider steps
+/*
+  Zoom slider and zoom to fit controls for GraphView
+*/
 
+import React from 'react';
+import FaExpand from 'react-icons/lib/fa/expand';
 
+const steps = 100; // Slider steps
 
-function makeStyles(primary){
-  return {
-    controls: {
-      position: 'absolute',
-      bottom: '30px',
-      left: '15px',
-      zIndex: 100,
-      display: 'grid',
-      gridTemplateColumns: 'auto auto',
-      gridGap: '15px',
-      alignItems: 'center'
-    },
-    sliderWrapper: {
-      backgroundColor: 'white',
-      color: primary,
-      border: `solid 1px lightgray`,
-      padding: '6.5px',
-      borderRadius: '2px'
-    },
-    slider: {
-      position: 'relative',
-      top: '6px',
-      marginLeft: 5,
-      marginRight: 5,
-      cursor: 'pointer'
-    },
-    button: {
-      backgroundColor: 'white',
-      color: primary,
-      border: `solid 1px lightgray`,
-      outline: 'none',
-      width: 31,
-      height: 31,
-      borderRadius: '2px',
-      cursor: 'pointer'
-    }
-  }
+type IGraphControlProps = {
+  maxZoom?: number;
+  minZoom?: number;
+  zoomLevel: number;
+  zoomToFit: (event: SyntheticMouseEvent<HTMLButtonElement>) => void;
+  modifyZoom: (delta: number) => boolean;
 }
 
+class GraphControls extends React.Component<IGraphControlProps> {
+  static defaultProps = {
+    maxZoom: 1.5,
+    minZoom: 0.15
+  };
 
-
-class GraphControls extends Component {
-
-  constructor(props) {
+  constructor(props: IGraphControlProps) {
     super(props);
-    this.state = {
-      styles: makeStyles(props.primary)
-    }
   }
 
   // Convert slider val (0-steps) to original zoom value range
-  sliderToZoom(val){
-    return ((val) * (this.props.maxZoom - this.props.minZoom)/steps ) + this.props.minZoom
+  sliderToZoom(val: number) {
+    const { minZoom, maxZoom } = this.props;
+    return val * ((maxZoom || 0) - (minZoom || 0)) / steps + (minZoom || 0);
   }
 
   // Convert zoom val (minZoom-maxZoom) to slider range
-  zoomToSlider(val){
-    return (val - this.props.minZoom) * steps/(this.props.maxZoom - this.props.minZoom)
-  }
-
-  // Center graph-view on contents of svg > view
-  zoomToFit(){
-    this.props.zoomToFit();
+  zoomToSlider(val: number) {
+    const { minZoom, maxZoom } = this.props;
+    return (val - (minZoom || 0)) * steps / ((maxZoom || 0) - (minZoom || 0));
   }
 
   // Modify current zoom of graph-view
-  zoom(e){
-    let sliderVal = e.target.value;
-    let zoomLevelNext = this.sliderToZoom(sliderVal);
-    let delta = zoomLevelNext-this.props.zoomLevel;
-
-    if( zoomLevelNext <= this.props.maxZoom && zoomLevelNext >= this.props.minZoom){
-      this.props.modifyZoom(delta)
+  zoom = (e: any) => {
```

Comment:
Like you mentioned, I do think it'd be good to create a task and update `any`s to a proper types (Think in this case you would use `SyntheticEvent<HTMLDivElement>`)

Reply:
Thanks. See #76.